java-checks/src/main/java/org/sonar/java/checks/regex/UnusedGroupNamesCheck.java

```diff
@@ -92,7 +94,7 @@ protected void checkRegex(RegexParseResult regexForLiterals, ExpressionTree meth
 
   private void checkGroupUsage(MethodInvocationTree mit, KnownGroupsCollector knownGroups) {
     String methodName = ExpressionUtils.methodName(mit).name();
-    if ("group".equals(methodName)) {
+    if (Arrays.asList(NAMES_OF_METHODS_WITH_GROUP_ARGUMENT).contains(methodName)) {
```

Comment:
We should duplicate the `NAMES_OF_METHODS_WITH_GROUP_ARGUMENT` as a list in a static field to avoid the unnecessary operation of initializing a new list on every call to the `checkGroupUsage`

Reply:
Good point, it's fixed